### PR TITLE
Replace password - return a session

### PR DESCRIPTION
**Pull request by asafshen**
## Description
related to https://github.com/descope/etc/issues/3800
Change `Descope.password.replace(...)` return value with authentication info

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -438,6 +438,70 @@ code the app produces.
...
+    "user-refresh-jwt"
+)
+```
+
+#### Replace Password
+
+To replace a user's password by providing their current password:
+
+```kotlin
+val descopeSession = Descope.password.replace(
```
Comment (itaihanski) [inline README.md:487]:
```suggestion
val authResponse = Descope.password.replace(
```

File: README.md [hunk 2]
```diff
@@ -438,6 +438,70 @@ code the app produces.
...
+    )
+)
+```
+
+#### Sign In with Password
+
+Authenticate an existing user using a password:
+
+```kotlin
+val descopeSession = Descope.password.signIn(
```
Comment (itaihanski) [inline README.md:464]:
```suggestion
val authResponse = Descope.password.signIn(
```

File: README.md [hunk 3]
```diff
@@ -438,6 +438,70 @@ code the app produces.
...
+### Password Authentication
+
+Authenticate users using a password.
+
+#### Sign Up with Password
+
+To create a new user that can later sign in with a password:
+
+```kotlin
+val descopeSession = Descope.password.signUp(
```
Comment (itaihanski) [inline README.md:450]:
```suggestion
val authResponse = Descope.password.signUp(
```

**Review by itaihanski — COMMENTED** [review]
Good!

File: descopesdk/src/main/java/com/descope/internal/routes/Password.kt [hunk 4]
```diff
@@ -31,10 +31,10 @@
         update(loginId, newPassword, refreshJwt)
     }
 
-    override suspend fun replace(loginId: String, oldPassword: String, newPassword: String) =
-        client.passwordReplace(loginId, oldPassword, newPassword)
+    override suspend fun replace(loginId: String, oldPassword: String, newPassword: String): AuthenticationResponse =
+        client.passwordReplace(loginId, oldPassword, newPassword).convert()
 
-    override fun replace(loginId: String, oldPassword: String, newPassword: String, callback: (Result<Unit>) -> Unit) = wrapCoroutine(callback) {
+    override fun replace(loginId: String, oldPassword: String, newPassword: String, callback: (Result<AuthenticationResponse>) -> Unit) = wrapCoroutine(callback) {
```
[the remark this answers is not in the document]
Comment (shilgapira) [reply descopesdk/src/main/java/com/descope/internal/routes/Password.kt:37]: Who asked for this? 😂

**Review by itaihanski — APPROVED** [review]

File: descopesdk/src/main/java/com/descope/internal/routes/Password.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (asafshen) [reply descopesdk/src/main/java/com/descope/internal/routes/Password.kt:37]: lol not sure! I think there are more lines like this, let me have a look 😄

**Pull request merged**